### ISSUE TITLE
ReviveClientConnection.Dispose() NullReferenceException when closing a WampChannel<TMessage> violently.

### ISSUE THREAD

**Issue by DonaldRecord** [Code-Sharp/WampSharp#150]
In the instance I saw this happening, I had a `WampChannel<TMessage>` (where TMessage is of type `Newtonsoft.Json.Linq.JToken`) that failed to open due to the target machine refusing the connection. For what I'm working on, that is a valid case - there may not be any connections for it use at that point in time. For this case, we have a "reconnector" class that will attempt reconnect. 

However, if it never acquires the connection and `WampChannel<TMessage>.Close()` is called, I get a `NullReferenceException`. It is calling `ReviveClientConnection.Dispose()` and the private `mConnection` member is null. Here is the top three lines of the call stack.

```
at WampSharp.Core.Listener.ReviveClientConnection`1.Dispose()
at WampSharp.V2.Client.WampChannel`1.Close()
at My.Custom.Class.WampReconnector.Dispose()
```

If any more information is needed, please let me know. Thank you.

**Comment by DonaldRecord** [comment]
Also, forgot to mention this; when I use the `WampChannel<JToken>.Close(string, GoodbyeDetails);` overload, I do not get the NullReferenceException.

Is this a viable workaround for calling 'Close()'? Are there any bad/"smelly" repercussions to doing it this way?
```
if (_channel.RealmProxy.Monitor.IsConnected)
                _channel.Close();
```

**Comment by darkl** [comment]
This issue has an easy fix, but I have been recently experiencing trouble with [TeamCity](http://teamcity.codebetter.com) CI. So release is dependent on TeamCity's resurgent.

I think your current workaround is fine.

Elad

**Comment by darkl** [comment]
TeamCity is sort of back. Released v1.2.5.23-beta version to NuGet. Sorry for the wait.
